sqlglot/parser.py

```diff
@@ -1225,6 +1225,10 @@ class Parser(metaclass=_Parser):
         **dict.fromkeys(("BINDING", "COMPENSATION", "EVOLUTION"), tuple()),
     }
 
+    PROCEDURE_OPTIONS: OPTIONS_TYPE = {}
+
+    EXECUTE_AS_OPTIONS: OPTIONS_TYPE = dict.fromkeys(("CALLER", "SELF", "OWNER"), tuple())
```

Comment:
Curious why these were left in the base class whereas `PROCEDURE_OPTIONS` where moved to T-SQL?

Reply:
I left them in the base class because that is where the parsing is happening. I could also just move everything to TSQL, but I was also planning on giving support to Snowflake stored procedure `EXECUTE AS` which is almost the same, so it would be useful to have that available on the parent class since I would be dealing with more than one dialect. Let me know what you think.

Reply:
Sounds good, thanks for clarifying.